### PR TITLE
Added responsive form 

### PR DESCRIPTION
**Pull request by mykhailo-diakovych**
(advantages: ordering inputs by grid-area, handling different window width, easy to configure)

## Summary by CodeRabbit

- **New Features**
  - Introduced new properties (`label`, `hint`, `isRequired`, `errorMessage`, `labelIcon`, `tooltipText`, `tooltipOptions`) to enhance form components with additional user guidance and validation feedback.
  - Expanded form components to include a full-width option for better layout control.
  - Added default country codes to the phone input for improved internationalization support.
  - Implemented a new `ResponsiveSmartForm` component for dynamic form creation with responsive layouts.
  - Created new storybook entries to showcase the `ResponsiveSmartForm` component with various field types and responsive grid layouts.

- **Enhancements**
  - Updated `OcTimePicker` to use dynamic imports for better performance.
  - Improved `ResponsiveSmartFormField` to dynamically render input fields and manage visibility and disabled states.
  - Enhanced `ResponsiveSmartFormGridLayout` with responsive grid styles for various breakpoints.

- **Refactor**
  - Streamlined export statements for better module organization.

- **Style**
  - Adjusted form builder layout to increase the number of columns for medium-sized screens.
  - Updated form element classes for consistent styling across components.

- **Bug Fixes**
  - Corrected initial value assignment in `OcFormBuilder` to prevent potential issues with form data handling.
  - Fixed conditional rendering in `OcTimePicker` to ensure the dropdown is displayed correctly.

- **Documentation**
  - Added grid layout configurations to sample data files for better documentation of responsive design capabilities.

### CODE REVIEW

**Comment by kewcoder** [comment]
@mykhailo-diakovych this feature like form builder right, Can we combine it with form builder? 
Screenshot 2023-11-16 at 23 11 52
form builder can also be used for testing all form components, you can see some components have errors, please help test and fix other input forms.

**Comment by kewcoder** [comment]
@mykhailo-diakovych this feature like form builder right, Can we combine it with form builder? 
Screenshot 2023-11-16 at 23 11 52
form builder can also be used for testing all form components, you can see some components have errors, please help test and fix other input forms.

**Review by kewcoder — CHANGES_REQUESTED** [review]
@mykhailo-diakovych i have add responsive form to builder using grid and col-span, If there are features missing in the form builder, help fix and improve. so we don't need responsiveSmartForm.

**Comment by kewcoder** [comment]
Screenshot 2023-11-17 at 22 21 15

 is there an example of a complex layout in figma. please make an example of using this feature, and why we need it, why not use the responsive tailwind css class ? @mykhailo-diakovych  cc @spacemonkey92

File: packages/@orchidui-vue/src/data/JsonForm.sample.js [hunk 1]
```diff
@@ -114,13 +123,9 @@ const SampleJsonForm = [
...
     },
   },
-  {
-    name: "toggle_field",
-    type: "Toggle",
-    className: "md:col-span-2",
-  },
   {
     name: "radio_group",
+    key: "radio_group",
```
Comment (kewcoder) [inline packages/@orchidui-vue/src/data/JsonForm.sample.js:128]: Can we use the name as the key.

**Review by kewcoder — APPROVED** [review]
It's a good idea, I appreciate it.

File: packages/@orchidui-vue/src/data/JsonForm.sample.js (hunk 1 above)
Comment (mykhailo-diakovych) [reply packages/@orchidui-vue/src/data/JsonForm.sample.js:128]: I think we could but what we do if name was an array?

Screenshot 2023-11-20 at 10 51 24

Comment (kewcoder) [reply packages/@orchidui-vue/src/data/JsonForm.sample.js:128]: https://github.com/hit-pay/orchid/pull/80/commits/5e1829f0dc46ef16f9c7272c1275011829d195b9  @mykhailo-diakovych What if it's like this? Is this good or bad code.

Comment (mykhailo-diakovych) [reply packages/@orchidui-vue/src/data/JsonForm.sample.js:128]: > [5e1829f](https://github.com/hit-pay/orchid/commit/5e1829f0dc46ef16f9c7272c1275011829d195b9) @mykhailo-diakovych What if it's like this? Is this good or bad code.

Yes, it is fine

Comment (kewcoder) [reply packages/@orchidui-vue/src/data/JsonForm.sample.js:128]: i will merge this PR, good idea for complex form builder

**Pull request merged**